frontend/src/component/common/NewConstraintAccordion/ConstraintsList/ViewableConstraintsList.tsx

```diff
@@ -0,0 +1,72 @@
+import { forwardRef } from 'react';
+import { styled } from '@mui/material';
+import type { IConstraint } from 'interfaces/strategy';
+import produce from 'immer';
+import useUnleashContext from 'hooks/api/getters/useUnleashContext/useUnleashContext';
+import { useWeakMap } from 'hooks/useWeakMap';
+import { constraintId } from 'component/common/LegacyConstraintAccordion/ConstraintAccordionList/createEmptyConstraint';
+import { ConstraintsList } from 'component/common/ConstraintsList/ConstraintsList';
+import { ConstraintAccordionView } from 'component/common/NewConstraintAccordion/ConstraintAccordionView/ConstraintAccordionView';
+import type {
+    IConstraintsListProps as IViewableConstraintsListProps,
+    IConstraintsListRef as IViewableConstraintsListRef,
+    IConstraintsListItemState as IViewableConstraintsListItemState,
+} from './ConstraintsListUtils';
+import { useConstraintsList } from './ConstraintsListUtils';
+
+export type { IViewableConstraintsListProps, IViewableConstraintsListRef };
+
+export const viewableConstraintsListId = 'viewableConstraintsListId';
+
+export const useViewableConstraintsList = useConstraintsList;
+
+const StyledContainer = styled('div')({
+    width: '100%',
+    display: 'flex',
+    flexDirection: 'column',
+});
+
+export const ViewableConstraintsList = forwardRef<
+    IViewableConstraintsListRef | undefined,
+    IViewableConstraintsListProps
+>(({ constraints, setConstraints }, ref) => {
+    const { context } = useUnleashContext();
+    const state = useWeakMap<IConstraint, IViewableConstraintsListItemState>();
+
+    const onEdit =
```

Comment:
I could not get rid of these yet, as the problem is that these are used very deep in some components, but its a start.

Reply:
Yup, I noticed when I was looking. But I'm hopuful for later 😄